tests/integration/tables/windows_firewall_rules.cpp

```diff
@@ -0,0 +1,242 @@
+
+  rule = r;
+  r.enabled = true;
+  rules.push_back(rule);
+
+  rule = r;
+  r.enabled = false;
+  rules.push_back(rule);
+
+  rule = r;
+  r.direction = NET_FW_RULE_DIR_OUT;
+  rules.push_back(rule);
+
+  rule = r;
+  r.protocol = NET_FW_IP_VERSION_V6;
+  rules.push_back(rule);
+
+  rule = r;
+  r.protocol = NET_FW_IP_VERSION_ANY;
+  rules.push_back(rule);
+
+  rule = r;
+  r.profileBitmask = NET_FW_PROFILE2_DOMAIN;
+  rules.push_back(rule);
+
+  rule = r;
+  r.profileBitmask = NET_FW_PROFILE2_PUBLIC;
+  rules.push_back(rule);
+
+  rule = r;
+  r.profileBitmask = NET_FW_PROFILE2_PRIVATE;
+  rules.push_back(rule);
+
+  rule = r;
+  r.profileBitmask = NET_FW_PROFILE2_DOMAIN | NET_FW_PROFILE2_PUBLIC;
+  rules.push_back(rule);
+
+  rule = r;
+  r.profileBitmask = NET_FW_PROFILE2_PUBLIC | NET_FW_PROFILE2_PRIVATE;
+  rules.push_back(rule);
+
+  rule = r;
+  r.localAddresses = "*";
+  rules.push_back(rule);
+
+  rule = r;
+  r.remoteAddresses = "LocalSubnet";
+  rules.push_back(rule);
+
+  rule = r;
+  r.localPorts = "*";
+  rules.push_back(rule);
+
+  rule = r;
+  r.remotePorts = "*";
+  rules.push_back(rule);
+
+  rule = r;
+  r.icmpTypesCodes = "8:*";
+  rules.push_back(rule);
+  return rules;
+}
+
+void validateRendered(const tables::FirewallRule& rule, Row& row) {
+  ASSERT_EQ(rule.name, row["name"]);
+  ASSERT_EQ(rule.appName, row["app_name"]);
+  switch (rule.action) {
+  case NET_FW_ACTION_BLOCK:
+    ASSERT_EQ("Block", row["action"]);
+    break;
+  case NET_FW_ACTION_ALLOW:
+    ASSERT_EQ("Allow", row["action"]);
+    break;
+  default:
+    ASSERT_EQ("", row["action"]);
+    break;
+  }
+
+  ASSERT_EQ(INTEGER(rule.enabled), row["enabled"]);
+
+  switch (rule.direction) {
+  case NET_FW_RULE_DIR_IN:
+    ASSERT_EQ("In", row["direction"]);
+    break;
+  case NET_FW_RULE_DIR_OUT:
+    ASSERT_EQ("Out", row["direction"]);
+    break;
+  default:
+    ASSERT_EQ("", row["direction"]);
+    break;
+  }
+
+  switch (rule.protocol) {
+  case NET_FW_IP_PROTOCOL_TCP:
+    ASSERT_EQ("TCP", row["protocol"]);
+    break;
+  case NET_FW_IP_PROTOCOL_UDP:
+    ASSERT_EQ("UDP", row["protocol"]);
+    break;
+  case NET_FW_IP_PROTOCOL_ANY:
+    ASSERT_EQ("Any", row["protocol"]);
+    break;
+  default:
+    ASSERT_EQ("", row["protocol"]);
+    break;
+  }
+
+  ASSERT_EQ(rule.localAddresses, row["local_addresses"]);
+  ASSERT_EQ(rule.remoteAddresses, row["remote_addresses"]);
+  ASSERT_EQ(rule.localPorts, row["local_ports"]);
+  ASSERT_EQ(rule.remotePorts, row["remote_ports"]);
+  ASSERT_EQ(rule.icmpTypesCodes, row["icmp_types_codes"]);
+
+  ASSERT_EQ(INTEGER(bool(rule.profileBitmask & NET_FW_PROFILE2_DOMAIN)),
+            row["profile_domain"]);
+  ASSERT_EQ(INTEGER(bool(rule.profileBitmask & NET_FW_PROFILE2_PRIVATE)),
+            row["profile_private"]);
+  ASSERT_EQ(INTEGER(bool(rule.profileBitmask & NET_FW_PROFILE2_PUBLIC)),
+            row["profile_public"]);
+}
+
+} // namespace
+
+TEST_F(windows_firewall_rules, test_firewall_rules_render) {
```

Comment:
Unit tests are located near the implementation; in this case, the path is the following: `osquery/tables/networking/tests/CMakeLists.txt`

Reply:
moved